lib/instrumentation/index.js

```diff
@@ -115,6 +116,8 @@ const IITM_MODULES = {
   https: { instrumentImportMod: true },
   ioredis: { instrumentImportMod: false },
   knex: { instrumentImportMod: false },
+  // XXX: check this
+  // mongodb: { instrumentImportMod: false },
```

Comment:
Note for reviewer: this is a work in progress though I do not mind to have it in a separate PR.

Reply:
Adding ESM in a separate PR would be helpful. Keeping this PR to a small change for using AsyncResource.bind might be helpful to compare with an equivalent OTel change.